R/loadFromMetaphlan.R

```diff
@@ -137,6 +137,24 @@ loadFromMetaphlan <- function(
             altExp(tse, rank) <- se_objects[[rank]]
         }
     }
+    # Set taxonomy ranks using .set_taxonomy_ranks
+    if(set.ranks) {
+        #If set.ranks == FALSE, skip the function by returning NULL
+        if(is.null(rowData(tse))) {
+            warning("rowData not found in the provided object.")
+        } else {
+            # character values from rowData, If none found, give warning and skip
+            rowDataChars <- rowData(tse)[,sapply(rowData(tse), is.character)]
```

Comment:
Does this cause problem when there is only one character column? You could add `drop = FALSE`